cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -828,74 +838,74 @@ createDelegateKeys :: FilePath -> Word -> ExceptT ShelleyGenesisCmdError IO ()
 createDelegateKeys dir index = do
   liftIO $ createDirectoryIfMissing False dir
   runGenesisKeyGenDelegate
-        (VerificationKeyFile $ dir </> "delegate" ++ strIndex ++ ".vkey")
-        coldSK
-        opCertCtr
+        (File @OfVerificationKey $ dir </> "delegate" ++ strIndex ++ ".vkey")
+        (onlyOut coldSK)
```

Comment:
To use an `InOut` file for an `Out` operation, we can cast it.  This works on all file types regardless of content type.